File: Selenium/Project/EditUserInfo.py
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait


def edit_info(web_driver, web_wait):
    web_wait.until(expected_conditions.element_to_be_clickable((By.ID, "menu_pim_viewMyDetails")))
    web_driver.find_element(By.ID, "menu_pim_viewMyDetails").click()
    web_driver.find_element(By.ID, "btnSave").click()
    web_wait.until(expected_conditions.element_to_be_clickable((By.ID, "personal_txtEmpFirstName")))
    first_name = driver.find_element(By.ID, "personal_txtEmpFirstName")
    first_name.clear()
    first_name.send_keys("Anu")
    last_name = driver.find_element(By.ID, "personal_txtEmpLastName")
    last_name.clear()
    last_name.send_keys("Balan")
    driver.find_element(By.ID, "personal_optGender_1").click()
    driver.find_element(By.ID, "personal_optGender_2").click()
    dob = driver.find_element(By.ID, "personal_DOB")
    dob.clear()
    dob.send_keys("1994-04-08")
    dropdown = driver.find_element(By.ID, "personal_cmbNation")
    select = Select(dropdown)
    select.select_by_value("38")
    driver.find_element(By.ID, "btnSave").click()
    time.sleep(5)


with webdriver.Chrome() as driver:
    driver.get("http://alchemy.hguy.co/orangehrm")
    title = driver.title
    print(title)

    driver.find_element(By.ID, "txtUsername").send_keys("orange")
    driver.find_element(By.ID, "txtPassword").send_keys("orangepassword123")
    driver.find_element(By.ID, "btnLogin").click()
    assert driver.find_element(By.XPATH, "//*[@id = 'welcome']").is_displayed();
    print(driver.find_element(By.TAG_NAME, "h1").text)
    wait = WebDriverWait(driver, 5)
    edit_info(driver, wait)

    driver.close()

File: Selenium/Project/EditUserInfo.java
package HRM;

import java.time.Duration;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Parameters;
import org.testng.annotations.Test;

public class EditUserInfo {
	
	static WebDriver driver;
	static WebDriverWait wait;

	
	@BeforeClass
	public void setUp() {
		
		driver = new ChromeDriver();
		driver.manage().window().maximize();
		driver.get("http://alchemy.hguy.co:8080/orangehrm/symfony/web/index.php/auth/login");
		driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
		wait = new WebDriverWait(driver, Duration.ofSeconds(5));
		
	}
	
	@Test(priority =1)
	@Parameters({"username", "password"})
	public void login_page(String userName, String password) {
		driver.findElement(By.id("txtUsername")).sendKeys(userName);
		driver.findElement(By.id("txtPassword")).sendKeys(password);
		driver.findElement(By.id("btnLogin")).click();
		Boolean logged_in = driver.findElement(By.xpath("//*[@id = 'welcome']")).isDisplayed();
		Assert.assertTrue(logged_in);
	}
	
	
	@Test(priority =2)
	public void editInfo() throws InterruptedException {
		
		wait.until(ExpectedConditions.elementToBeClickable(By.id("menu_pim_viewMyDetails")));
		driver.findElement(By.id("menu_pim_viewMyDetails")).click();
		driver.findElement(By.id("btnSave")).click();
		wait.until(ExpectedConditions.elementToBeClickable(By.id("personal_txtEmpFirstName")));
		WebElement firstName = driver.findElement(By.id("personal_txtEmpFirstName"));
		firstName.clear();
		firstName.sendKeys("Anu");
		WebElement lastName = driver.findElement(By.id("personal_txtEmpLastName"));
		lastName.clear();
		lastName.sendKeys("Balan");
		driver.findElement(By.id("personal_optGender_1")).click();
		driver.findElement(By.id("personal_optGender_2")).click();
		WebElement dob = driver.findElement(By.id("personal_DOB"));
		dob.clear();
		dob.sendKeys("1994-04-08");
		WebElement dropdown = driver.findElement(By.id("personal_cmbNation"));
		Select select = new Select(dropdown);
		select.selectByValue("36");
		driver.findElement(By.id("btnSave")).click();
	}
	
	@AfterClass
	public void tearDown() {
		driver.quit();
	}
	
	
}